src-tauri/src/api.rs

```diff
@@ -144,17 +171,65 @@ pub async fn generate_remote(
 ) -> ChoreoResult<TrajFile> {
     let remote_resources = app_handle.state::<RemoteGenerationResources>();
     use choreo_core::generation::remote::remote_generate_parent;
-    remote_generate_parent(&remote_resources, project, traj, handle).await
+    debug_result!(remote_generate_parent(&remote_resources, project, traj, handle).await);
 }
 
 #[tauri::command]
 pub fn cancel_remote_generator(app_handle: tauri::AppHandle, handle: i64) -> ChoreoResult<()> {
     let remote_resources = app_handle.state::<RemoteGenerationResources>();
-    remote_resources.kill(handle)
+    debug_result!(remote_resources.kill(handle));
 }
 
 #[tauri::command]
 pub fn cancel_all_remote_generators(app_handle: tauri::AppHandle) {
     let remote_resources = app_handle.state::<RemoteGenerationResources>();
     remote_resources.kill_all();
 }
+
+#[tauri::command]
+pub fn open_diagnostic_file(project: ProjectFile, trajs: Vec<TrajFile>) -> ChoreoResult<()> {
+    tracing::debug!("Opening diagnostic file");
+    let log_lines = if let Some(dir) = dirs::state_dir().map(|d| d.join("logs")) {
```

Comment:
The more standard place to put this is somewhere like state_dir()/choreo/log.

Reply:
How does this work on Windows? As far as I know, state_dir() just returns None on Windows.

Reply:
It doesn't. We should fix that.

Reply:
That seems like a good place to put it.

Reply:
I feel like the diagnostic zips should go somewhere under temp.